manylinux_v2_centos/Dockerfile

```
@@ -0,0 +1,95 @@
+ARG BASE_IMAGE=nvidia/cuda:11.8.0-devel-centos7
+FROM ${BASE_IMAGE}
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ENV PYENV_ROOT="/pyenv" \
+    PATH="/pyenv/bin:/pyenv/shims:$PATH"
+
+RUN yum update -y && yum install -y epel-release wget gcc zlib-devel bzip2 bzip2-devel readline-devel sqlite sqlite-devel xz xz-devel libffi-devel curl git ncurses-devel numactl numactl-devel openssh-clients libcudnn8-devel zip blas-devel lapack-devel protobuf-compiler autoconf automake libtool centos-release-scl scl-utils cmake && yum clean all
+
+# Install tools that have to be installed after the previous install completes.
+RUN yum remove -y git && yum install -y https://packages.endpointdev.com/rhel/7/os/x86_64/endpoint-repo.x86_64.rpm && yum install -y git jq devtoolset-11 && yum remove -y endpoint-repo
+
+# Install latest openssl (can't use the package manager because CentOS 7 stopped after openssl 1.0)
+RUN pushd tmp && \
+    wget https://ftp.openssl.org/source/openssl-1.1.1k.tar.gz && \
+    tar -xzvf openssl-1.1.1k.tar.gz &&\
+    cd openssl-1.1.1k && \
+    ./config --prefix=/usr --openssldir=/etc/ssl --libdir=lib no-shared zlib-dynamic && \
+    make && \
+    make install && \
+    popd
+
+# Install sccache
+ARG SCCACHE_VERSION=0.4.2
+ARG ARCH="x86_64"
+
+RUN curl -o /tmp/sccache.tar.gz \
+        -L "https://github.com/mozilla/sccache/releases/download/v${SCCACHE_VERSION}/sccache-v${SCCACHE_VERSION}-${ARCH}-unknown-linux-musl.tar.gz" && \
+        tar -C /tmp -xvf /tmp/sccache.tar.gz && \
+        mv "/tmp/sccache-v${SCCACHE_VERSION}-${ARCH}-unknown-linux-musl/sccache" /usr/bin/sccache && \
+        chmod +x /usr/bin/sccache
+
+# Install ucx
+ARG UCX_VERSION=1.13.0
```

Do we want to update?
```suggestion
ARG UCX_VERSION=1.14.0
```

---

That's a good question. I think we could probably update this. I can test that out in a build.

---

Since I copied most versions out of the old images, I'm going to leave them as is for now. I would like to do an update, but at the moment I think it makes sense to avoid rocking that boat in order to get this merged and move on to CUDA 12 wheels. Once those are working and stable I'll come back to the updates.